src/Shared/AspireStore.cs

```diff
@@ -0,0 +1,176 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using System.Security.Cryptography;
+using IdentityModel;
+using Microsoft.Extensions.SecretManager.Tools.Internal;
+
+namespace Aspire.Hosting.Utils;
+
+internal sealed class AspireStore : KeyValueStore
```

Comment:
I think we will want to make this API public because any integration (including outside of dotnet/aspire) will want to write to this folder.